idseq/locations.py

```diff
@@ -0,0 +1,150 @@
+"""Module for handling location metadata and geosearching."""
+
+import random
+import requests
+import threading
+import time
+# For Python2 compatibility
+from builtins import input
+
+MAX_GEOSEARCH_ATTEMPTS = 3
+MAX_GEOSEARCH_THREADS = 5
+
+
+def geosearch_and_set_csv_locations(base_url, headers, csv_data, project_id):
+    """Automatically geosearch CSV collection locations for matches."""
+    raw_names = get_raw_locations(csv_data)
+
+    matched_locations = fetch_location_matches(raw_names, base_url, headers)
+    if len(matched_locations) > 0:
+        confirm_location_matches(matched_locations)
+
+    set_location_matches(csv_data, matched_locations)
+    print_location_matches(csv_data, base_url, project_id)
+    return csv_data
+
+
+def get_raw_locations(csv_data):
+    raw_names = set()
+    for metadata in csv_data.values():
+        for field_name, value in metadata.items():
+            if field_name.lower() in ["collection_location", "collection location"]:
```

Comment:
I wonder if we should have this fields from server?
Maybe add it to a visible constant on top at least ( I see you repeat it further down)?

Reply:
👍